WalletWasabi/Nito/AsyncEx/AsyncMutex.cs

```diff
@@ -401,21 +401,35 @@ public static async Task WaitForAllMutexToCloseAsync()
 				}
 			}
 
+			var lastLog = DateTimeOffset.MinValue;
+			var lastNumberOfAliveMutexes = 0;
+
 			while (true)
 			{
+				KeyValuePair<string, AsyncMutex>[] asyncMutexes = null;
 				lock (AsyncMutexesLock)
 				{
-					bool stillRunning = AsyncMutexes.Any(am => am.Value.IsAlive);
-					if (!stillRunning)
-					{
-						return;
-					}
-					Logger.LogDebug($"Waiting for: {string.Join(", ", AsyncMutexes.Where(am => am.Value.IsAlive).Select(m => m.Value.ShortName))}.");
+					asyncMutexes = AsyncMutexes.ToArray();
 				}
-				await Task.Delay(200).ConfigureAwait(false);
-				if (DateTime.Now - start > TimeSpan.FromSeconds(60))
+
+				int numberOfAliveMutexes = asyncMutexes.Count(am => am.Value.IsAlive);
+				if (numberOfAliveMutexes == 0)
+				{
+					return;
+				}
+
+				// Log every 10 seconds or status change.
+				if ((DateTimeOffset.UtcNow - lastLog) > TimeSpan.FromSeconds(10) || lastNumberOfAliveMutexes != numberOfAliveMutexes)
+				{
+					Logger.LogDebug($"Waiting for: {string.Join(", ", asyncMutexes.Where(am => am.Value.IsAlive).Select(m => m.Value.ShortName))}.");
+					lastNumberOfAliveMutexes = numberOfAliveMutexes;
+					lastLog = DateTimeOffset.UtcNow;
+				}
+
+				await Task.Delay(500).ConfigureAwait(false);
```

Comment:
500ms is a foreverness for a computer. Please reduce it to 100ms as that seems instantenous for a user.

Reply:
Done.